### PR TITLE
libtecla: test added

### PR DESCRIPTION
**Pull request by lordqwerty**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/libtecla.rb [hunk 1]
```diff
@@ -14,8 +14,36 @@ class Libtecla < Formula
   end
 
   def install
-    ENV.deparallelize
-    system "./configure", "--prefix=#{prefix}", "--mandir=#{man}"
-    system "make", "install"
+    ENV.deparallelize do
```
Comment (fxcoudert) [inline Formula/libtecla.rb]: `ENV.deparallelize` was already present, and making it into a block is a no-op.

File: Formula/libtecla.rb [hunk 2]
```diff
@@ -14,8 +14,36 @@ class Libtecla < Formula
...
+      system "./configure", "prefix=#{prefix}", "mandir=#{man}"
```
Comment (fxcoudert) [inline Formula/libtecla.rb]: This is a traditional `configure`, the options need to be of the form `--prefix`

File: Formula/libtecla.rb [hunk 3]
```diff
@@ -14,8 +14,36 @@ class Libtecla < Formula
...
+      system "make", "install"
+    end
+  end
+
+  test do
+    (testpath/"test.c").write <<-EOS.undent
+      #include <stdio.h>
```
Comment (fxcoudert) [inline Formula/libtecla.rb]: `stdio.h` and `string.h` should not be necessary

**Review by fxcoudert — CHANGES_REQUESTED** [review]
Thanks for your contribution! The test is a very good idea.

**Comment by fxcoudert** [comment]
Thanks, can you squash the commits into one, with commit message `libtecla: add test`

File: Formula/libtecla.rb [hunk 4]
```diff
@@ -18,4 +18,29 @@ def install
...
+        char *line;
+        GetLine *gl;
+
+        setlocale(LC_CTYPE, "");
+
+        gl = new_GetLine(1024, 2048);
+        if(!gl)
+          return 1;
+
+        gl = del_GetLine(gl);
```
Comment (maxim-belkin) [inline Formula/libtecla.rb]: what value is `gl` assigned to in case of a failure? can you add something similar to lines 35-36 above?

Comment (fxcoudert) [inline Formula/libtecla.rb]: It always returns NULL, says the doc.

Comment (lordqwerty) [inline Formula/libtecla.rb]: I don't think this matters. 35 - 36 cover failure enough for test. Line 38 deleting as not doing anything of significance.

Comment (maxim-belkin) [inline Formula/libtecla.rb]: then just don't assign this to `gl`

**Comment by lordqwerty** [comment]
I've done something horribly wrong... *gulp*

**Comment by fxcoudert** [comment]
You have picked up an unrelated commit.

File: Formula/libtecla.rb (hunk 4 above)
Comment (fxcoudert) [inline Formula/libtecla.rb]: It's fine…

**Comment by lordqwerty** [comment]
Sorry for the trouble.

**Comment by maxim-belkin** [comment]
How about making this a tad more compact?
```ruby
  test do
    (testpath/"test.c").write <<-EOS.undent
      #include <locale.h>
      #include <libtecla.h>

      int main(int argc, char *argv[]) {
        GetLine *gl;
        setlocale(LC_CTYPE, "");
        gl = new_GetLine(1024, 2048);
        if (!gl) return 1;
        return 0;
      }
    EOS

    system ENV.cc, "test.c", "-L#{lib}", "-ltecla", "-o", "test"
    system "./test"
  end
```

**Comment by lordqwerty** [comment]
Please let me know if you would like further changes. If not I believe ready to merge.

**Comment by maxim-belkin** [comment]
I think it is worth combining lines 28 and 30

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thanks @lordqwerty for your contribution!

**Pull request merged**